src/backend/metal/src/command.rs

```diff
@@ -3088,6 +3084,8 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
                                 }
                                 target_offset.buffers += 1;
                             }
+                        } else if layout.content.contains(native::DescriptorContent::DYNAMIC_BUFFER) {
+                            dynamic_offset_iter.next();
```

Comment:
What's the reason we don't increment `*offset` in this case too? i.e. in comparison to `bind_graphics_descriptor_sets`, where increase `offset` regardless of the stage flags

Reply:
There are 2 offsets in play:
  - data offset that points to our actual descriptors stored by the pool. This is bumped regardless of the stage, just a line below
  - target offset, which is the Metal binding slot. It's only increased for a stage that a descriptor binds to, and it happens the same way in graphics/compute

This branch you highlighted is only suspicious at first because of the changed order of conditions:
  - graphics goes for each descriptor, then for each sub-type, then for each stage
  - compute goes for each descriptor, then for compute stage, then for each sub-type

So I added this little branch to advance the dynamic offsets properly... maybe worth rewriting it to be totally like in graphics, will just going to be a tiny bit slower since there is only one stage to check for.

Reply:
Thanks, I think I understand this part. I'll attempt to rephrase, my confusion is:
- In `bind_graphics_descriptor_sets` we do `*offset += *dynamic_offset_iter.next().unwrap().borrow() as u64;` for the current `layout`, regardless of whether or not we have vertex or fragment flags
- In `bind_compute_descriptor_sets` we do `*offset += *dynamic_offset_iter.next().unwrap().borrow() as u64;` for the current `layout` only if we have a compute flag, otherwise `dynamic_offset_iter.next();`

So I'm trying to understand why the buffer's offset is not updated in the latter compute case, because it seems inconsistent – this all seems to occur in the `layouts` loop, so I thought the logic would match.

Reply:
Ah, I see your confusion better now. Thanks for explanation!
The offset only needs to be modified if we actually use it (to write into our resource state, or to issue a command). In `bind_compute_descriptor_sets` if there is no `COMPUTE` flag, we don't use that offset for anything, and thus we just skip it.